tests/test_compiler_entrypoints.py

```diff
@@ -309,3 +310,55 @@ def verify_anchor(flnr, s, e, docs):
     verify_anchor(23, 11, 15, "this is a test entity")
     verify_anchor(24, 29, 30, None)
     verify_anchor(24, 11, 22, None)
+
+
+def test_constructor_with_inferred_namespace(snippetcompiler):
+    module: str = "tests"
+    target_path = os.path.join(os.path.dirname(__file__), "data", "modules", module, "model", "_init.cf")
+
+    snippetcompiler.setup_for_snippet(
+        """
+    import mod1
+    import tests
+    entity A:
+    end
+
+    A.mytest [1] -- tests::Test [1]
+
+    A(mytest = Test())
+    """,
+        autostd=False,
+    )
+
+    compiler = Compiler()
+    (statements, blocks) = compiler.compile()
+    sched = scheduler.Scheduler()
+
+    def check_anchor(
+        range_list: Sequence[tuple[Range, Range]],
+        range_source: tuple[str, int, int, int],
+        range_target: tuple[str, int, int, int],
+    ) -> bool:
+        file1, lnr1, start_char1, end_char1 = range_source
+        file2, lnr2, start_char2, end_char2 = range_target
+        for range1, range2 in range_list:
+            # Check if the properties match for both ranges
+            if (
+                range1.file == file1
+                and range1.lnr == lnr1
+                and range1.start_char == start_char1
+                and range1.end_char == end_char1
+                and range2.file == file2
+                and range2.lnr == lnr2
+                and range2.start_char == start_char2
+                and range2.end_char == end_char2
+            ):
+                return True
+
+        return False
```

Comment:
I'm not sure I understand what you mean: We iterate over all the tuples present in the anchormap looking for the one that corresponds to the constructor we are interested in. If we find it we are happy, otherwise it is missing and we are sad. This test doesn't care about the other anchors present in the anchormap. I could verify the length of the anchormap to make sure he have the right number of anchors. I'll add that.

Reply:
Oh I see, I completely missed the point !
Then I think you could simplify this by creating 2 `Range´ objects instead of tuples and asserting:
```python
range_source = Range(...)
range_target = Range(...)
assert (range_source, range_target) in anchormap
```

Reply:
indeed, way better, thanks!